src/test/java/io/wisetime/connector/utils/DurationCalculatorTest.java

```diff
@@ -0,0 +1,157 @@
+ */
+class DurationCalculatorTest {
+
+  private FakeEntities fakeEntities = new FakeEntities();
+
+  @Test
+  void config_defaultCalculator() {
+    final User user = fakeEntities.randomUser().experienceWeightingPercent(50);
+
+    final TimeGroup timeGroup = fakeEntities
+        .randomTimeGroup()
+        .totalDurationSecs(120)
+        .user(user);
+
+    final DurationCalculator.Result result = DurationCalculator.of(timeGroup).calculate();
+
+    assertThat(result.getTotalDuration())
+        .isEqualTo(60)
+        .as("Default calculator configuration should use the TimeGroup duration " +
+            "and apply the user's experience rating");
+  }
+
+  @Test
+  void config_useDurationFrom_timeGroup() {
+    final TimeGroup timeGroup = fakeEntities
+        .randomTimeGroup()
+        .totalDurationSecs(120);
+
+    final DurationCalculator.Result result = DurationCalculator
+        .of(timeGroup)
+        .disregardExperienceRating()
+        .useDurationFrom(DurationSource.TIME_GROUP)
+        .calculate();
+
+    assertThat(result.getTotalDuration())
+        .isEqualTo(120)
+        .as("Duration should be taken from the time group");
+  }
+
+  @Test
+  void config_useDurationFrom_timeRows() {
+    final TimeRow timeRow1 = fakeEntities.randomTimeRow().durationSecs(70);
+    final TimeRow timeRow2 = fakeEntities.randomTimeRow().durationSecs(80);
+
+    final TimeGroup timeGroup = fakeEntities
+        .randomTimeGroup()
+        .totalDurationSecs(0)
+        .timeRows(ImmutableList.of(timeRow1, timeRow2));
+
+    final DurationCalculator.Result result = DurationCalculator
+        .of(timeGroup)
+        .disregardExperienceRating()
+        .useDurationFrom(DurationSource.SUM_TIME_ROWS)
+        .calculate();
+
+    assertThat(result.getTotalDuration())
+        .isEqualTo(150)
+        .as("Duration should be taken from the time rows");
+  }
+
+  @Test
+  void config_useExperienceRating() {
+    final User user = fakeEntities.randomUser().experienceWeightingPercent(50);
+
+    final TimeGroup timeGroup = fakeEntities
+        .randomTimeGroup()
+        .totalDurationSecs(120)
+        .user(user);
+
+    final DurationCalculator.Result result = DurationCalculator
+        .of(timeGroup)
+        .useExperienceRating()
+        .useDurationFrom(DurationSource.TIME_GROUP)
+        .calculate();
+
+    assertThat(result.getTotalDuration())
+        .isEqualTo(60)
+        .as("The user's experience rating should be taken into account");
+  }
+
+  @Test
+  void config_disregardExperienceRating() {
+    final User user = fakeEntities.randomUser().experienceWeightingPercent(50);
+
+    final TimeGroup timeGroup = fakeEntities
+        .randomTimeGroup()
+        .totalDurationSecs(120)
+        .user(user);
+
+    final DurationCalculator.Result result = DurationCalculator
+        .of(timeGroup)
+        .disregardExperienceRating()
+        .useDurationFrom(DurationSource.TIME_GROUP)
+        .calculate();
```

Comment:
Please review the API and let me know if naming makes sense.

Reply:
Nice and easy to understand API